From 68ebe8659857d8beaac5cde2c53540dd0dc8e883 Mon Sep 17 00:00:00 2001
From: Ranjan Dutta <ranjan.dutta@intel.com>
Date: Tue, 21 Sep 2021 12:30:19 +0800
Subject: [PATCH 02/30] Revert "drm/msm/dpu: make
 dpu_hw_ctl_clear_all_blendstages clear necessary LMs"

This reverts commit 7125705623f9ebef50f547d4e324daa889317f18.
---
 drivers/gpu/drm/msm/disp/dpu1/dpu_hw_ctl.c | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/drivers/gpu/drm/msm/disp/dpu1/dpu_hw_ctl.c b/drivers/gpu/drm/msm/disp/dpu1/dpu_hw_ctl.c
index f8c7100a8acb..758c355b4fd8 100644
--- a/drivers/gpu/drm/msm/disp/dpu1/dpu_hw_ctl.c
+++ b/drivers/gpu/drm/msm/disp/dpu1/dpu_hw_ctl.c
@@ -340,12 +340,10 @@ static void dpu_hw_ctl_clear_all_blendstages(struct dpu_hw_ctl *ctx)
 	int i;
 
 	for (i = 0; i < ctx->mixer_count; i++) {
-		enum dpu_lm mixer_id = ctx->mixer_hw_caps[i].id;
-
-		DPU_REG_WRITE(c, CTL_LAYER(mixer_id), 0);
-		DPU_REG_WRITE(c, CTL_LAYER_EXT(mixer_id), 0);
-		DPU_REG_WRITE(c, CTL_LAYER_EXT2(mixer_id), 0);
-		DPU_REG_WRITE(c, CTL_LAYER_EXT3(mixer_id), 0);
+		DPU_REG_WRITE(c, CTL_LAYER(LM_0 + i), 0);
+		DPU_REG_WRITE(c, CTL_LAYER_EXT(LM_0 + i), 0);
+		DPU_REG_WRITE(c, CTL_LAYER_EXT2(LM_0 + i), 0);
+		DPU_REG_WRITE(c, CTL_LAYER_EXT3(LM_0 + i), 0);
 	}
 }
 
-- 
2.27.0

